classes/local/step/flow_logic_case.php

```diff
@@ -192,6 +192,7 @@ public function next($caller) {
                     $this->passed = false;
                 }
                 $value = $this->input->current();
+                $steps = $caller->step->engine->get_variables()['steps'];
```

Comment:
Why not allow for all variables?
```suggestion
                $variables = $caller->step->engine->get_variables();
                $variables['record'] = $value;
```
And then pass this through instead.

Reply:
And perhaps call it $stepsvariables to not confuse them with actual steps.

Reply:
@jaypha There wouldn't be any more `$steps` variable so the suggestion is not relevant here?